### DISCUSSION TITLE
Import of Joplin documents

### DISCUSSION THREAD

**Discussion by Waxwood** [TriliumNext/Trilium#10159] [Q&A]
Questions.

1. Is there an easy way of importing a Joplin note database?
I'd like to switch to Trilium because of its excellent tree-structure layout and WYSIWYG document viewer. I cannot see the value of Joplin's markup- clumsy and slow. Its lack of a tree structure and clumsy interface wastes screen space and makes finding notes slow and clumsy. 

2. Is there an easy way to import multiple WORD documents. I'd like to export my note database from MyInfo, another Personal Information Manager. 

3. Can you improve the webclipper to offer options like Joplin?

4. Please include as many developers as possible to ensure longevity. This is an amazing application.

**Comment by zadam** [comment]
Hi,

1.  You can export Joplin notes to Markdown (assuming Joplin supports that), zip them and then import the ZIP file into Trilium.
2.  Word documents are not supported, the only way is to manually copy paste from open Word to the Trilium - formatting should be partially preserved.
3.  That's not really actionable request, if you want some improvements you need to describe what features you want (ideally create one issue per feature).
4.  Contributions are welcome, but in practice it's just me doing the majority of work.

**Comment by Waxwood** [comment]
Hi Zadam, 
Thank you for your reply. 
Outstanding work on Trilium. It really is the best of all open source note-takers. I hope you can build a team so the project gathers momentum and users. I can help with documentation if you need assistance. 
I'll create a detailed note on how the webclipper can be improved.

**Comment by BillDett** [comment]
Excellent tool @zadam- thanks for your hard work on this. I'm also looking at moving over from Joplin- I see the Markdown import works just fine but it does not preserve the creation/modification dates (even if I include them as front matter in the exported md files from Joplin). Is there any elegant way to batch update the date/time stamps on these notes after import? Or do I need to hack the database directly? Any advice on how to do this would be appreciated.

**Comment by bewille** [comment]
Hi,
I would like to change from Joplin to Trilium for reasons of backup, versioning etc.
I imported a huge markdown structure from Joplin (using Export Markdown + Front matter). Works fine but leaves all ressources in a directory "ressources". Of course all inks are broken. The links in Markdown  read /../../_ressources/xyz in the notes. Any idea where to place the directory in the Trilium structure or how to change the links in the Joplin-Markdown to refer to something again?

**Comment by D4rkT1m0** [comment]
any update for the previous question regarding the "resource" folder?

**Comment by bewille** [comment]
Sorry, I have to tell you I gave in and moved to another not-taking application.

Best regards
Bernd

**Comment by Nriver** [comment]
> any update for the previous question regarding the "resource" folder?

If you want to import Joplin to Trilium, you can try https://github.com/Nriver/trilium-py. It will handle the pictures for you.

**Comment by D4rkT1m0** [comment]
Thank you nriver, I tried trilium-py and it worked just great. 
I didn't even know python, but I learn how to use it to install this on my synology nas. 
It worked as planned, ignoring the _ressource folder from
My Joplin markdown export.

**Comment by Nriver** [comment]
That's great to hear! I'm glad trilium-py worked well for your needs, and kudos to you for learning how to use Python to get it set up on your Synology NAS.
